src/MagnumPlugins/WebPImporter/Test/WebPImporterTest.cpp

```diff
@@ -0,0 +1,411 @@
+                '\x5d', '\x4b', '\x2f', '\x80',
+                '\x5d', '\x4b', '\x2f', '\xff',
+                '\x5d', '\x4b', '\x2f', '\x80',
+
+                '\xb7', '\xa8', '\xb0', '\xff',
+                '\xb7', '\xa8', '\xb0', '\xff',
+                '\xb7', '\xa8', '\xb0', '\xff',
+
+                '\x5d', '\x58', '\xa2', '\x80',
+                '\x5d', '\x58', '\xa2', '\xff',
+                '\x5d', '\x58', '\xa2', '\x80',
+            }), TestSuite::Compare::Container);
+            break;
+
+        /* Same image with 0% image quality (lossy bitstream) */
+        case 3:
+            CORRADE_COMPARE_AS(image->data(), Containers::arrayView<char>({
+                '\x2c', '\x31', '\x12', '\x80',
+                '\x2c', '\x31', '\x12', '\xff',
+                '\x2c', '\x31', '\x12', '\x80',
+
+                '\xcb', '\xcd', '\xbe', '\xff',
+                '\xcb', '\xcd', '\xbe', '\xff',
+                '\xcb', '\xcd', '\xbe', '\xff',
+
+                '\x5e', '\x5c', '\x6c', '\x80',
+                '\x5e', '\x5c', '\x6c', '\xff',
+                '\x5e', '\x5c', '\x6c', '\x80',
+            }), TestSuite::Compare::Container);
+            break;
+    }
+}
+
+void WebPImporterTest::openMemory() {
+    auto&& data = OpenMemoryData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+    Containers::Optional<Containers::Array<char>> memory = Utility::Path::read(Utility::Path::join(WEBPIMPORTER_TEST_DIR, "rgb-lossless.webp"));
+    CORRADE_VERIFY(memory);
+    CORRADE_VERIFY(data.open(*importer, *memory));
+
+    Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+    CORRADE_VERIFY(image);
+    CORRADE_COMPARE(image->size(), Vector2i(3, 3));
+    CORRADE_COMPARE(image->format(), PixelFormat::RGB8Unorm);
+
+    CORRADE_COMPARE_AS(image->pixels<UnsignedByte>().asContiguous(), Containers::arrayView<UnsignedByte>({
+        1, 2, 3,
+        4, 5, 6,
+        7, 8, 9
+    }), TestSuite::Compare::Container);
+}
+
+void WebPImporterTest::openTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(WEBPIMPORTER_TEST_DIR, "penguin-lossless.webp")));
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(WEBPIMPORTER_TEST_DIR, "penguin-lossless.webp")));
+
+    /* Shouldn't crash, leak or anything */
+}
+
+void WebPImporterTest::importTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("WebPImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(WEBPIMPORTER_TEST_DIR, "penguin-lossless.webp")));
+
+    /* Verify that everything is working the same way on second use */
+    {
+        Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+        CORRADE_VERIFY(image);
+        CORRADE_COMPARE(image->size(), (Vector2i{386, 395}));
```

Comment:
You may want to chose a smaller version of this file, reasoning being that with the many test files in the repository, it would take a long time to clone the repository.
Maybe something like 12x11 pixels could work, you can just rescale this one. (Maybe wait for input from @mosra, he might have a clearer idea of what this image should be)

Especially also nature.png

Reply:
Yes, testing with one of the tiny images used already in the other tests above should be fine, no need to add another file just for the `openTwice()` and `importTwice()` tests.

Reply:
I have removed unnecessary images and scaled nature.png down to a very small image.